main/src/ca/uwaterloo/flix/language/phase/Codegen.scala

```diff
@@ -1095,7 +1095,7 @@ object Codegen {
           case BinaryOperator.NotEqual => (IF_ICMPEQ, FCMPG, DCMPG, IF_ACMPEQ, IFEQ)
         }
         e1.tpe match {
-          case Type.Unit | Type.Str | Type.Enum(_, _) if o == BinaryOperator.Equal || o == BinaryOperator.NotEqual =>
+          case Type.Unit | Type.Str if o == BinaryOperator.Equal || o == BinaryOperator.NotEqual =>
             // Unit, String, and Enum can be reference compared for equality.
```

Comment:
Please update the comment. Not just this one, but also the method documentation.